pkg/commands/oscommands/cmd_obj_builder.go

```diff
@@ -10,12 +10,10 @@ import (
 )
 
 type ICmdObjBuilder interface {
-	// New returns a new command object based on the string provided
-	New(cmdStr string) ICmdObj
+	// NewFromArgs takes a slice of strings like []string{"git", "commit"} and returns a new command object.
+	NewFromArgs(args []string) ICmdObj
```

Comment:
Now that `New` no longer exists, it would make sense to rename `NewFromArgs` to `New`, since it's the primary way to create an instance.

Reply:
Agreed